src/Model/Item.php

```diff
@@ -1445,7 +1445,24 @@ public static function isValid(array $item)
 			Logger::notice('Causer is ignored by user', ['causer-link' => $item['causer-link'], 'uid' => $item['uid'], 'item-uri' => $item['uri']]);
 			return false;
 		}
-		
+
+		$toplevelOwnerId = null;
+		if ($item['parent-uri'] != $item['uri']) {
+			$topLevelUriId   = ItemURI::getIdByURI($item['parent-uri']);
+			$toplevelItem    = self::selectFirst(['owner-link'], ['uri-id' => $topLevelUriId]);
```

Comment:
You could simply check for a parent where `wall` is set to `true`. In the `uid` field of that post you would have the original thread starter.

Reply:
Should that check be recursive, like `getParentData`?

Reply:
There is no need for recursion. When you have got the real `parent-uri-id` then you can make a single query for a post with the `uri-id` set to the `parent-uri-id` and `wall` set to `true`. The `uid` is then the correct user.